pyresample/future/spherical/point.py

```diff
@@ -0,0 +1,151 @@
+        gl.ylabels_right = False
+
+    return ax
+
+
+def check_lon_validity(lon):
+    """Check longitude validity."""
+    if np.any(np.isinf(lon)):
+        raise ValueError("Longitude values can not contain inf values.")
+
+
+def check_lat_validity(lat):
+    """Check latitude validity."""
+    if np.any(np.isinf(lat)):
+        raise ValueError("Latitude values can not contain inf values.")
+    if np.any(np.logical_or(lat > np.pi / 2, lat < -np.pi / 2)):
+        raise ValueError("Latitude values must range between [-pi/2, pi/2].")
+
+
+class SPoint(SCoordinate):
+    """Spherical Point.
+
+    The ``lon`` and ``lat`` coordinates must be provided in radians.
+    """
+
+    def __init__(self, lon, lat):
+        lon = np.asarray(lon)
+        lat = np.asarray(lat)
+        if lon.size > 1 or lat.size > 1:
+            raise ValueError("Use SMultiPoint to define multiple points.")
+        check_lon_validity(lon)
+        check_lat_validity(lat)
+
+        super().__init__(lon, lat)
+
+    @property
+    def vertices(self):
+        """Return SPoint vertices in a ndarray of shape [1,2]."""
+        return np.array([self.lon, self.lat])[None, :]
+
+    def plot(self, ax=None, **plot_kwargs):
+        """Plot the SPoint using Cartopy."""
+        ax = _plot(self.vertices, ax=ax, **plot_kwargs)
+        return ax
+
+    def to_shapely(self):
+        """Convert the SPoint to a shapely Point (in lon/lat degrees)."""
+        from shapely.geometry import Point
+        point = Point(*np.rad2deg(self.vertices[0]))
+        return point
+
+
+class SMultiPoint(SCoordinate):
+    """Create SPoint or SMultiPoint object."""
+
+    def __new__(cls, lon, lat):
+        """Create SPoint or SMultiPoint object."""
+        lon = np.asarray(lon)
+        lat = np.asarray(lat)
+
+        # If a single point, define SPoint
+        if lon.ndim == 0:
+            return SPoint(lon, lat)  # TODO: TO BE DEFINED
+        # Otherwise a SMultiPoint
+        else:
+            return super().__new__(cls)
+
+    def __init__(self, lon, lat):
+        super().__init__(lon, lat)
+
+    @property
+    def vertices(self):
+        """Return SMultiPoint vertices in a ndarray of shape [n,2]."""
+        return np.vstack((self.lon, self.lat)).T
+
+    def __eq__(self, other):
+        """Check equality."""
+        return np.allclose(self.lon, other.lon) and np.allclose(self.lat, other.lat)
+
+    def __str__(self):
+        """Get simplified representation of lon/lat arrays in degrees."""
+        vertices = np.rad2deg(np.vstack((self.lon, self.lat)).T)
+        return str(vertices)
+
+    def __repr__(self):
+        """Get simplified representation of lon/lat arrays in degrees."""
+        vertices = np.rad2deg(np.vstack((self.lon, self.lat)).T)
+        return str(vertices)
+
+    def plot(self, ax=None, **plot_kwargs):
+        """Plot the SMultiPoint using Cartopy."""
+        ax = _plot(self.vertices, ax=ax, **plot_kwargs)
+        return ax
```

Comment:
This is the same as in SPoint and could thus be moved in the base class

Reply:
This has been moved together with vertices properties to SCoordinate.